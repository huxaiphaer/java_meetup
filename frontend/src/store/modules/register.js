import axios from "axios";
import { BASE_URL } from "../../utils/Urls";

export default {
  namespaced: true,
  state: {
    snackbar: {
      snackbar_visibility: false,
      color: "",
      timeout: 0,
      message: ""
    },
    loader: false
  },

  mutations: {
    reset_snackbar(state) {
      state.snackbar.message = "";
      state.snackbar.color = "";
      state.snackbar.timeout = 0;
      state.snackbar.snackbar_visibility = false;
    },

    toast_snackbar_on_success(state) {
      state.snackbar.message = "User registered up Successfully";
      state.snackbar.color = "#4CAF50";
      state.snackbar.timeout = 1000;
      state.snackbar.snackbar_visibility = true;
    },
    toast_snackbar_on_error(state, message) {
      state.snackbar.message = message;
      state.snackbar.color = "#9c4130";
      state.snackbar.timeout = 1000;
      state.snackbar.snackbar_visibility = true;
    },

    loader_on(state) {
      state.loader = true;
    },

    loader_off(state) {
      state.loader = false;
    }
  },
  actions: {
    toast_snackbar_on_error({ commit }, message) {
      commit("toast_snackbar_on_error", message);
    },
    reset_snackbar({ commit }) {
      commit("reset_snackbar");
    },
    registerUsers({ commit }, data) {
      commit("loader_on");
      axios
        .post(`${BASE_URL}/api/v1/users`, data)
        .then(response => {
          commit("loader_on");
          if (response.status == 201) {
            setTimeout(() => {
              commit("reset_snackbar");
            }, 4000);
            commit("toast_snackbar_on_success");
            commit("loader_off");
          } else {
            setTimeout(() => {
              commit("toast_snackbar_on_error", response);
            }, 4000);
            commit("loader_off");
          }
        })
        .catch(error => {
          commit("toast_snackbar_on_error", error.response.data.errors || error.response.data.message)
          commit("loader_off");
        });
    }
  },
  getters: {
    snackbar_status(state) {
      return state.snackbar;
    },
    loader_status(state) {
      return state.loader;
    }
  }
};
