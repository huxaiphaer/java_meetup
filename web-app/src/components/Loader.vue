<template>
    <v-progress-linear
            value="15"
            active="loader_status"
            color="primary"
            :indeterminate="loader_status"
            top="top"
            light="light"
            v-show="loader_status"
    ></v-progress-linear>
</template>

<script>
    import { mapGetters } from 'vuex';

    export default {
        name: 'Loader',
        computed: {
            ...mapGetters('register', ['loader_status']),
        },
    };
</script>

<style scoped></style>
