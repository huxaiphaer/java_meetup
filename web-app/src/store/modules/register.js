import axios from "axios";
import {BASE_URL} from "../../utils/Urls";

export default {
    namespaced: true,
    state: {
        snackbar: {
            snackbar_visibility: false,
            color: "",
            timeout: 0,
            message: ""
        },
        loader: false,
    },

    mutations: {
        reset_snackbar(state) {
            state.snackbar.message = '';
            state.snackbar.color = '';
            state.snackbar.timeout = 0;
            state.snackbar.snackbar_visibility = false;
        },

        toast_snackbar_on_success(state) {
            state.snackbar.message = 'User registered up Successfully';
            state.snackbar.color = '#4CAF50';
            state.snackbar.timeout = 1000;
            state.snackbar.snackbar_visibility = true;
        },
        toast_snackbar_on_error(state, message) {
            state.snackbar.message = message;
            state.snackbar.color = '#9c4130';
            state.snackbar.timeout = 1000;
            state.snackbar.snackbar_visibility = true;
        },

        loader_on(state) {
            state.loader = true;
        },

        loader_off(state) {
            state.loader = false;
        },


    },
    actions: {
        toast_snackbar_on_error({commit}, message) {
            commit('toast_snackbar_on_error', message);
        },
        reset_snackbar({commit}) {
            commit('reset_snackbar');
        },
        registerUsers({commit}, data) {
            commit('loader_on');
            axios
                .post(`${BASE_URL}/api/v1/users`, data)
                .then(response => {
                    if (response.status == 200) {
                        setTimeout(() => {
                            commit('reset_snackbar');
                        }, 2000);
                        commit('toast_snackbar_on_success');
                        commit('loader_off');

                    } else if (response.status == 400) {
                        console.log(" whats here  " + response)
                        commit('toast_snackbar_on_error', response);
                        commit('loader_off');
                    }
                })
                .catch(error => {
                    console.log(" error " + error.response.data)
                    error.response.data.errors.map((t) => {
                        commit('toast_snackbar_on_error', t);
                    })
                    commit('loader_off');
                });
        }
    },
    getters: {
        snackbar_status(state) {
            return state.snackbar;
        },
        loader_status(state) {
            return state.loader;
        },

    }
};
