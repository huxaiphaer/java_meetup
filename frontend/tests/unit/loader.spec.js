import register from "@/store/modules/register";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import Vuex from "vuex";
import Vuetify from "vuetify";
import Loader from "../../src/components/Loader";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe("Loader.vue", () => {
  let vuetify;
  let store;
  beforeEach(() => {
    store = new Vuex.Store({ modules: { register } });
  });

  it("should have a Loader component and match snapshot", () => {
    const wrapper = shallowMount(Loader, {
      vuetify,
      store,
      localVue
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
