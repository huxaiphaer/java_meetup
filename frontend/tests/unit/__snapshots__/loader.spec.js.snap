// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.vue should have a Loader component and match snapshot 1`] = `<v-progress-linear-stub color="primary" top="true" value="15" light="true" active="loader_status" buffervalue="100" height="4" style="display: none;"></v-progress-linear-stub>`;
